### ISSUE TITLE
Support for non-expiring Azure DevOps credentials

### ISSUE THREAD

**Issue by giuliov** [tfsaggregator/aggregator-cli#14]
Current version has only support for PAT, which can last, at most, a year.

**Comment by pswetz** [comment]
Its seems the issue here is secure storage am I correct?

**Comment by giuliov** [comment]
Yes (currently PAT is saved in the Function configuration), and more: ideally the Function instance should be perpetually connected, removing the hassle of manually refreshing expiring tokens.

**Comment by BobSilent** [comment]
@giuliov 
please correct me if I am going into wrong direction: Is this about these 2 topics?

1) storage location of secrets (PAT)
- use Azure Key Vault for storage in Azure Function Context
- use windows vault for local execution 

2) support additional login possibilities
- support running with dedicated provided username:password?
- support OAuth2

**Comment by giuliov** [comment]
@BobSilent it is more 2b in order to avoid managing PAT
I explored the topic a while ago, but I was only able to find solutions for:

- the client scenario, like a WPF App or a Web site, implying user interaction;
- AAD authentication using Graph API, which may apply or not (not very familiar with this, but I guess we should test with a DevOps Organization backed by MSA accounts and one using AAD)

The ideal solution would be similar to registering a build agent: you authenticate once then some kind of token is exchanged and the agent is trusted.

PS you can use [Slack](https://slackin-tfsaggregator.herokuapp.com/) for chatting and discussion.
